Crashlytics/Crashlytics/Controllers/FIRCLSMetricKitManager.m

```diff
@@ -0,0 +1,311 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+
+#import "Crashlytics/Crashlytics/Components/FIRCLSCrashedMarkerFile.h"
+#import "Crashlytics/Crashlytics/Controllers/FIRCLSManagerData.h"
+#import "Crashlytics/Crashlytics/Controllers/FIRCLSMetricKitManager.h"
+#import "Crashlytics/Crashlytics/Helpers/FIRCLSCallStackTree.h"
+#import "Crashlytics/Crashlytics/Helpers/FIRCLSFile.h"
+#import "Crashlytics/Crashlytics/Models/FIRCLSExecutionIdentifierModel.h"
+#import "Crashlytics/Crashlytics/Models/FIRCLSInternalReport.h"
+#import "Crashlytics/Crashlytics/Public/FirebaseCrashlytics/FIRCrashlyticsReport.h"
+
+@interface FIRCLSMetricKitManager () {
+  NSLock *_mutex;
+  FBLPromise *_metricKitDataAvailable;
+  FIRCLSExistingReportManager *_existingReportManager;
+  FIRCLSFileManager *_fileManager;
+  FIRCLSManagerData *_managerData;
```

Comment:
For all these variables (except mutex), we're tryna move away from this style and move towards using `@property`. Check out `FIRCLSExistingReportManager` as an example. For usage of these variables, keep using `_varName` in the `init` method, but for the rest of the instances replace it with `self.varName`